File: rust/src/schemas/stats_response.rs
use std::collections::HashMap;

use serde::Serialize;

use crate::models::{priority::Priority, ticket_status::TicketStatus};

#[derive(Serialize)]
pub struct StatsResponse {
    pub by_status: HashMap<TicketStatus, i32>,
    pub by_priority: HashMap<Priority, i32>,
}

File: springboot/src/main/java/com/example/tickets/dto/StatsResponse.java
package com.example.tickets.dto;

import com.example.tickets.model.Priority;
import com.example.tickets.model.TicketStatus;

import java.util.Map;

// Rust parallel: struct StatsResponse { by_status: HashMap<TicketStatus, u64>, by_priority: HashMap<Priority, u64> }
public record StatsResponse(
        Map<TicketStatus, Long> byStatus,
        Map<Priority, Long> byPriority
) {}
